db.go

```diff
@@ -851,7 +882,13 @@ func (r *Rows) initScan(t reflect.Type) error {
 func (r *Rows) scanValue() error {
 	// Clear out our value object in preparation for the scan.
 	r.value.Set(r.zero)
-	return r.Rows.Scan(r.dest...)
+	if err := r.Rows.Scan(r.dest...); err != nil {
+		return err
+	}
+	for i, c := range r.convert {
+		c.set.Set(reflect.ValueOf(r.dest[i]).Elem().Convert(c.typ))
```

Comment:
I think the reflect package panics if a value can't be converted. We might want to return an error if it's unconvertible:
```go
src := reflect.ValueOf(r.dest[i]).Elem()
if !src.Type().ConvertibleTo(c.typ) {
        return fmt.Errorf("%s cannot be converted to type %s", src, c.typ)
}
c.set.Set(src.Convert(c.typ))
```